packages/sdk/observability/src/helpers/node-observability.ts

```diff
@@ -0,0 +1,152 @@
+//
+// Copyright 2022 DXOS.org
+//
+import yaml from 'js-yaml';
+import { existsSync, statSync } from 'node:fs';
+import { mkdir, readFile, writeFile } from 'node:fs/promises';
+import { join } from 'node:path';
+import { v4 as uuid, validate as validateUuid } from 'uuid';
+
+import type { Config } from '@dxos/client';
+import { log } from '@dxos/log';
+
+import { Observability, type Mode } from '../observability';
+
+/**
+ * Print telemetry banner once per installation.
+ */
+export const showObservabilityBanner = async (configDir: string, bannercb: (input: string) => void) => {
+  const path = join(configDir, '.telemetry-banner-printed');
+  if (existsSync(path)) {
+    return;
+  }
+  bannercb(
+    // eslint-disable-next-line no-multi-str
+    'Basic observability data will be sent to the DXOS team in order to improve the product. This includes \
+    performance metrics, error logs, and usage data. No personally identifiable information, other than your \
+    public key, is included with this data and no private data ever leaves your devices. To disable sending \
+    observability data, set the environment variable DX_DISABLE_TELEMETRY=true.',
+  );
+
+  await writeFile(path, '', 'utf-8');
+};
+
+export const getObservabilityState = async (configDir: string): Promise<PersistentObservabilityState> => {
+  // check whether configDir exists and if it's a directory
+
+  if (existsSync(configDir)) {
+    if (!statSync(configDir).isDirectory()) {
+      throw new Error(`Config directory ${configDir} exists but is not a directory`);
+    }
+  } else {
+    await mkdir(configDir, { recursive: true });
+  }
+
+  // named 'telemetry.yml' for backwards compatibility
```

Comment:
fwiw probably will hurt less to change this sooner rather than later

Reply:
will change the localforage key then too?